rfc/rfc-42/rfc-42.md

```diff
@@ -0,0 +1,220 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+# RFC-42: Implement Compaction/Clustering Service for Hudi
+
+## Proposers
+- @yuzhaojing
+
+## Approvers
+- @vinothchandar
+
+## Status
+JIRA: [https://issues.apache.org/jira/browse/HUDI-3016](https://issues.apache.org/jira/browse/HUDI-3016)
+
+## Abstract
+
+Hudi table needs compaction/clustering to rewrite data. Currently, schedule compaction/clustering job provides Three ways:
+
+- Inline, execute compaction/clustering job and writing job in the same application, perform the compaction/clustering job and writing job serially.
+
+- Async, execute compaction/clustering job and writing job in the same application, Async parallel execution of compaction/clustering job and write job.
+
+- Independent compaction/clustering job, execute an async compaction/clustering job of another application.
+
+With the increase in the number of HUDI tables, due to a lack of management capabilities, maintenance costs will become higher. This proposal is to implement an independent compaction/clustering Service to manage the Hudi compaction/clustering job.
```

Comment:
may I know why existing independent job https://github.com/apache/hudi/blob/master/hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieCompactor.java did not work out? interested to know any gaps.

Reply:
It can work now, but in large-scale job scenarios, if each job corresponds to an independent job, the entire system becomes difficult to manage, and it also needs to rely on an external scheduling system. So we need an independent service to manage these compaction jobs so that users can use independent compaction jobs more conveniently.

Reply:
+1